src/main/java/Duke.java

```
@@ -1,10 +1,211 @@
+						throw new DukeException(e);	
+					}
+					
+				break;
+
+			case "mark":
+				try{
+					markTask(Integer.parseInt(parm.get(1)), true);
+				}
+				catch (IndexOutOfBoundsException e) {
+					throw new DukeException(e);
+				
+				}
+				catch (NumberFormatException e) {
+					throw new DukeException(e);
+				
+				}
+				catch (Exception e) {
+					throw new DukeException(e);	
+				}
+					
+				break;
+			case "unmark":
+				try{
+					markTask(Integer.parseInt(parm.get(1)), false);
+				}
+				catch (IndexOutOfBoundsException e) {
+					throw new DukeException(e);
+				
+				}
+				catch (NumberFormatException e) {
+					throw new DukeException(e);
+				
+				}
+				catch (Exception e) {
+					throw new DukeException(e);	
+				}
+					
+				break;
+			
+			case "delete":
+				try{
+					deleteTask(Integer.parseInt(parm.get(1)));
+				}
+				catch (IndexOutOfBoundsException e) {
+					throw new DukeException(e);
+				
+				}
+				catch (NumberFormatException e) {
+					throw new DukeException(e);
+				
+				}
+				catch (Exception e) {
+					throw new DukeException(e);	
+				}
+
+				break;
+					
+			default:
+				throw new DukeException();
+
+		}
+
+	}	
+
+	public static void main(String[] args) throws DukeException{
+	Duke duke = new Duke();
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+	System.out.println("I will remember things now");
+
+	while(true) {
+		System.out.print('\n');
+		Scanner sc = new Scanner(System.in);
+		String in = sc.nextLine();
+
+		if (in.equals("bye")) {
+			System.out.println("No don't go!!");
+			break;
+		}
+		
+		line(in.length());
+		String[] parmArr = in.split("\\s+");
+		List<String> parm = Arrays.asList(parmArr);
+		try {
+			parseIn(parm);
+		}
+		catch (DukeException e) {
+			
```

I see that the DukeException will send a message to user when it is created, I believe adding a comment for this behaviour would be helpful for others to understand why there is an empty catch block.